test/integration/api/datasets.js

```diff
@@ -15,6 +15,330 @@ const { exhaust } = require(appRoot + '/lib/worker/worker');
+            body.should.be.a.Dataset();
+            body.name.should.equal('trees');
+            body.properties.should.eql([]);
+            body.approvalRequired.should.eql(true);
+          });
+      }));
+
+      it('should reject if creating a dataset that already exists', testService(async (service) => {
+        const asAlice = await service.login('alice');
+
+        await asAlice.post('/v1/projects/1/datasets')
+          .send({
+            name: 'trees'
+          })
+          .expect(200);
+
+        // Second time
+        await asAlice.post('/v1/projects/1/datasets')
+          .send({
+            name: 'trees'
+          })
+          .expect(409);
+      }));
+
+      it('should add label-only entity to dataset, all via API', testService(async (service) => {
+        const asAlice = await service.login('alice');
+
+        await asAlice.post('/v1/projects/1/datasets')
+          .send({
+            name: 'trees'
+          })
+          .expect(200);
+
+        await asAlice.post('/v1/projects/1/datasets/trees/entities')
+          .send({
+            uuid: '12345678-1234-4123-8234-123456789aaa',
+            label: 'Willow',
+            data: {}
+          })
+          .expect(200);
+
+        await asAlice.get('/v1/projects/1/datasets/trees/entities')
+          .then(({ body }) => {
+            body.length.should.equal(1);
+          });
+
+        const result = await asAlice.get('/v1/projects/1/datasets/trees/entities.csv')
+          .expect(200)
+          .then(r => r.text);
+
+        const isoRegex = /\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}Z/g;
+
+        result.match(isoRegex).should.have.length(1);
+
+        const withOutTs = result.replace(isoRegex, '');
+        withOutTs.should.be.eql(
+          '__id,label,__createdAt,__creatorId,__creatorName,__updates,__updatedAt,__version\n' +
+            '12345678-1234-4123-8234-123456789aaa,Willow,,5,Alice,0,,1\n'
+        );
+      }));
+
+      it('should log an event in the audit log for creating a new dataset', testService(async (service) => {
+        const asAlice = await service.login('alice');
+
+        await asAlice.post('/v1/projects/1/datasets')
+          .send({
+            name: 'trees'
+          })
+          .expect(200);
+
+        await asAlice.get('/v1/audits')
+          .set('X-Extended-Metadata', 'true')
+          .then(({ body: logs }) => {
+            logs[0].action.should.equal('dataset.create');
+            logs[0].actorId.should.equal(5);
+            logs[0].actee.should.be.a.Dataset();
+            logs[0].actee.name.should.equal('trees');
+            logs[0].details.properties.should.eql([]);
+          });
+      }));
+    });
+
+    describe('projects/:id/datasets/:dataset/properties POST', () => {
+      it('should reject user does not have dataset update access', testService(async (service) => {
+        const asChelsea = await service.login('chelsea');
+
+        await asChelsea.post('/v1/projects/1/datasets')
```

Comment:
This looks like the same test as above. I think this one needs to send a request to the .../properties endpoint.